ui/raidboss/data/07-dt/raid/r6s.ts

```diff
@@ -519,6 +563,186 @@ const triggerSet: TriggerSet<Data> = {
+      id: 'R6S Tempest Piece Spawn Collector',
+      type: 'AddedCombatant',
+      netRegex: { npcNameId: '13827', capture: true },
+      condition: (data) => data.cloudId === undefined,
+      preRun: (data, matches) => {
+        data.cloudId = matches.id;
+        const cloudDir =
+          Directions.output8Dir[Directions.addedCombatantPosTo8Dir(matches, 100, 100)];
+        switch (cloudDir) {
+          case 'dirNE':
+          case 'dirNW':
+          case 'dirS':
+            data.cloudPos = cloudDir;
+            break;
+          default:
+            console.log(
+              `R6S Tempest Piece Spawn Collector - Invalid cloud spawn direction ${cloudDir}`,
+            );
+            return;
+        }
+        const cloudX = parseFloat(matches.x);
+        const cloudY = parseFloat(matches.y);
+        data.cloudLastAngle = Math.atan2(cloudX - 100, cloudY - 100);
+        // For the initial spawn, the actor moves in place which causes a false positive firing
+        // of the detection trigger. Avoid it by setting new angle to last angle
+        data.cloudNewAngle = data.cloudLastAngle;
+      },
+      infoText: (data, _matches, output) =>
+        output.spawn!({ dir: output[data.cloudPos ?? 'unknown']!() }),
+      outputStrings: {
+        ...Directions.outputStrings8Dir,
+        spawn: {
+          en: 'Cloud spawning ${dir}',
+        },
+      },
+    },
+    {
+      id: 'R6S Tempest Piece Rotation Detector',
+      type: 'ActorMove',
+      netRegex: { capture: true },
+      condition: (data, matches) => {
+        if (matches.id !== data.cloudId)
+          return false;
+        return data.cloudId !== undefined && data.cloudLastAngle !== undefined &&
+          data.cloudNewAngle === undefined && data.cloudExplosionCount < 5;
+      },
+      preRun: (data, matches) => {
+        const cloudX = parseFloat(matches.x);
+        const cloudY = parseFloat(matches.y);
+        data.cloudNewAngle = Math.atan2(cloudX - 100, cloudY - 100);
+      },
+      infoText: (data, _matches, output) => {
+        const lastAngle = data.cloudLastAngle;
+        const newAngle = data.cloudNewAngle;
+        if (lastAngle === undefined || newAngle === undefined)
+          throw new UnreachableCode();
+        const cwRotation = (Math.PI + newAngle) < (Math.PI + lastAngle);
+        if (data.cloudPos === 'dirNE') {
+          data.cloudPos = cwRotation ? 'dirS' : 'dirNW';
+        } else if (data.cloudPos === 'dirNW') {
+          data.cloudPos = cwRotation ? 'dirNE' : 'dirS';
+        } else {
+          data.cloudPos = cwRotation ? 'dirNW' : 'dirNE';
+        }
+        return output.text!({
+          rot: output[cwRotation ? 'cw' : 'ccw']!(),
+          dir: output[data.cloudPos ?? 'unknown']!(),
+        });
+      },
+      outputStrings: {
+        ...Directions.outputStrings8Dir,
+        cw: Outputs.clockwise,
+        ccw: Outputs.counterclockwise,
+        text: {
+          en: 'Cloud rotating ${rot} towards ${dir}',
```

Comment:
Re: your Discord remark about this text, I don't know that we can really make it shorter without losing information. `Away from ${dir}` is more concise but doesn't help with getting to the safespot.

Reply:
Moving backwards through the thundercloud is a perfectly valid movement that I sometimes do as healer if that's the shorter path to my potential next spread spot. I'm not sold on the wording for this trigger at all, this is more "placeholder for a better phrasing" to me.

Reply:
The cloud only leaves one safe bridge opposite from it, so it might be clear enough to call something like, `Cloud ${dir}, Party ${safeDir}` or maybe `Safe bridge ${safeDir}`?

Reply:
That's a good point, "call the safe location" seems reasonable.

Reply:
These triggers are functional as-is, no issues from Tuesday reclear (across the 7 pulls it took us to reclear, people are still sloppy on mechanics after adds...)

I won't have time to update the cloud rotate trigger until this weekend most likely. I'm not sure if we want to merge these as-is and fix the cloud trigger text in a follow up or wait until Saturday for me to have free time again.

Reply:
I think if it's functional as the player would expect, it's probably fine to merge and then update when one of us has time.

Reply:
Instead of "Cloud rotating ${rot}", it can be shortened to something like "${rot} Cloud, ${safeSpot1}/${safeSpot2} Safe' .
It might need to be split up for outputs to support various strats, but that could be done in configuration.

Reply:
Additionally, safespot calls could just go to the headmarkered players, and the non-headmarked players could get a directional Bridge call. This would merge two triggers, reducing the current simultaneous call.